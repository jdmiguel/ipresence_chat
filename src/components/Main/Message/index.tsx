import React from 'react';
import { InView } from 'react-intersection-observer';

import { observerOptions } from '../../../helpers/utils';
import { IDisplayedMessage } from '../../../helpers/types';

interface IMessageProps {
  data: IDisplayedMessage;
  onUnreadMessage: (id: number) => void;
}

const Message: React.FC<IMessageProps> = ({
  data: {
    id,
    text,
    messageClasses,
    isUnread,
    hasIcon,
    iconClasses,
    iconName,
    date,
  },
  onUnreadMessage,
}) => (
  <InView
    {...observerOptions}
    as="div"
    className={messageClasses}
    onChange={(inView) => {
      if (inView && isUnread) {
        onUnreadMessage(id);
      }
    }}
  >
    <div className="message-content">
      <p>{text}</p>
      {hasIcon && (
        <i className={iconClasses && iconClasses}>{iconName && iconName}</i>
      )}
    </div>
    <div className="message-time">
      <p>{date}</p>
    </div>
  </InView>
);

export default Message;
