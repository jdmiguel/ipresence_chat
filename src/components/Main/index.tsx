import React from 'react';

const Main:React.FC = () => {
  return(
    <main>
      <div>
        <p>MAIN</p>
      </div>
    </main>
  )
}

export default Main;