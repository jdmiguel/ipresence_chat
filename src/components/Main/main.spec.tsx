import React from 'react';
import { render } from '@testing-library/react';

import Main from '.';

const MAIN_PROPS = {
  onScroll: (event: React.UIEvent) => () => {},
  onUnreadMessage: () => {},
  displayedMessages: [],
};

describe('Component: Main', () => {
  it('should render', () => {
    const { container } = render(
      <Main {...MAIN_PROPS} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});