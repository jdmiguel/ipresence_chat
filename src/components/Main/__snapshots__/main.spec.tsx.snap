// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Main should render 1`] = `
<main>
  <div />
</main>
`;
